packages/integration-sdk-runtime/src/logger/registerEventEmitters.ts

```diff
@@ -0,0 +1,117 @@
+import { IntegrationLogger } from '.';
+
+type EventType = 'multipleResolves'|'unhandledRejection'|'uncaughtException';
+
+type LifecycleErrorCallback = (err: Error, event: EventType) => void;
+
+// conforms to type MultipleResolveListener, see node_modules/@types/node/globals.d.ts
+type MultipleResolveListener = (type: 'resolve'|'reject', promise: Promise<any>, value: any) => void;
+
+function createMultipleResolveListener(callback: LifecycleErrorCallback): MultipleResolveListener {
+  return (type, promise, value) => {
+    if (type === 'reject') {
+      callback(value, 'multipleResolves');
+    }
+  };
+}
+
+// conforms to type UnhandledRejectionListener, see node_modules/@types/node/globals.d.ts
+type UnhandledRejectionListener = (reason: {} | null | undefined, promise: Promise<any>) => void;
+
+function createUnhandledRejectionListener(callback: LifecycleErrorCallback): UnhandledRejectionListener {
+  return (reason, promise) => {
+    callback(reason as Error, 'unhandledRejection');
+  };
+}
+
+// conforms to type UncaughtExceptionListener, see node_modules/@types/node/globals.d.ts
+type UncaughtExceptionListener = (error: Error) => void;
+
+function createUncaughtExceptionListener(callback: LifecycleErrorCallback): UncaughtExceptionListener {
+  return (error) => {
+    callback(error, 'uncaughtException');
+  };
+}
+
+interface RegisteredEventEmitters {
```

Comment:
Aren't these really registered event listeners? They all have the name `*Listener`. 🤔

Reply:
Agreed 😄  check the new commit 💯